Tests/AESGCMSpecTest.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author(s): Yan Peng
+-/
+import Arm.BitVec
+import Specs.AES
+import Specs.GCM
+
+-- Reference: https://csrc.nist.rip/groups/ST/toolkit/BCM/documents/proposedmodes/gcm/gcm-spec.pdf
+
+namespace AES256GCMSpecTest
+
+open BitVec
+
+def CIPH_128 : GCM.Cipher (n := 128) (m := 128) :=
+  AES.AES_encrypt (Param := AES.AES128KBR)
+
+def GCM_AE_128 : BitVec 128 → BitVec lv → BitVec p → BitVec a → (t : Nat)
+  → (BitVec p) × (BitVec t) := GCM.GCM_AE (m := 128) CIPH_128
+
+def GCM_AD_128 : BitVec 128 → BitVec lv → BitVec c → BitVec a → BitVec t
+  → Option (BitVec c) := GCM.GCM_AD (m := 128) CIPH_128
+
+def CIPH_256 : GCM.Cipher (n := 128) (m := 256) :=
+  AES.AES_encrypt (Param := AES.AES256KBR)
+
+def GCM_AE_256 : BitVec 256 → BitVec lv → BitVec p → BitVec a → (t : Nat)
+  → (BitVec p) × (BitVec t) := GCM.GCM_AE (m := 256) CIPH_256
+
+def GCM_AD_256 : BitVec 256 → BitVec lv → BitVec c → BitVec a → BitVec t
+  → Option (BitVec c) := GCM.GCM_AD (m := 256) CIPH_256
+
+-- Test Vector 0
+
+def K0 : BitVec 128 := 0
+def IV0 : BitVec 96 := 0
+def P0 : BitVec 0 := BitVec.nil
+def A0 : BitVec 0 := BitVec.nil
+def C0 : BitVec 0 := BitVec.nil
+def T0 : BitVec 128 := rev_elems 128 8 0x58e2fccefa7e3061367f1d57a4e7455a#128 (by decide) (by decide)
```

Comment:
A naive question: what is rev_elems?

Reply:
`rev_elems` is a function defined in Arm/Insts/Common.lean. See the following:
```
/-- Reverse the order of `esize`-bit elements in `x`.-/
def rev_elems (n esize : Nat) (x : BitVec n) (h₀ : esize ∣ n) (h₁ : 0 < esize) : BitVec n

example : rev_elems 4 4 0xA#4 (by decide) (by decide) = 0xA#4 := rfl
example : rev_elems 8 4 0xAB#8 (by decide) (by decide) = 0xBA#8 := rfl
example : rev_elems 8 4 (rev_elems 8 4 0xAB#8 (by decide) (by decide))
          (by decide) (by decide) = 0xAB#8 := by native_decide
```